src/map/attack.cpp

```diff
@@ -112,36 +112,20 @@ void CAttack::SetCritical(bool value, uint16 slot, bool isGuarded)
     }
 }
 
-/************************************************************************
- *                                                                      *
- *  Sets the guarded flag.                                              *
- *                                                                      *
- ************************************************************************/
-void CAttack::SetGuarded(bool isGuarded)
-{
-    m_isGuarded = isGuarded;
-}
-
 /************************************************************************
  *                                                                      *
  *  Gets the guarded flag.                                              *
  *                                                                      *
  ************************************************************************/
 bool CAttack::IsGuarded()
 {
-    m_isGuarded = attackutils::IsGuarded(m_attacker, m_victim);
-    if (m_isGuarded)
+    if (m_isGuarded.has_value())
     {
-        if (m_damageRatio > 1.0f)
-        {
-            m_damageRatio -= 1.0f;
-        }
-        else
-        {
-            m_damageRatio = 0;
-        }
+        return m_isGuarded.value();
     }
-    return m_isGuarded;
+
+    m_isGuarded.emplace(attackutils::IsGuarded(m_attacker, m_victim));
+    return m_isGuarded.value();
```

Comment:
Is there a reason we removed the entirety of m_damageRatio? I don't see where the damage reduction is taking place if this is removed

Reply:
Yes, this already being done in battleutils::GetDamageRatio
https://github.com/AirSkyBoat/AirSkyBoat/blob/staging/src/map/utils/battleutils.cpp#L3086-L3105

When gaurd was being recalc'd repeatedly before - The guard pdif correction was either being done before pdif was set (resulting in no impact) or could be done multiple times post pdif calculation.

So the work being done in IsGuarded() was at best superfolous, or at worst allowed critical guards to hit dmgRatio 0,